### PR TITLE
Database integrated in browser

### PR DESCRIPTION
**Pull request by annierusu**
Database is linked to the browser with the additional property for the card to be owned or wanted.

### CODE REVIEW

**Review by Cocograal — APPROVED** [review]
All good

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt
Comment (annierusu) [inline app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt:49]: possession should be owner for this add card

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardViewModel.kt
Comment (annierusu) [inline app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardViewModel.kt:31]: is this done?

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardFragment.kt
Comment (KelvinKappeler) [reply app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardFragment.kt:132]: In addition, the comment implies that this boolean should be changed to true if the user is connected. That's fine, but this function doesn't know the user. It's just a function that shows or not some buttons. I don't remember how we called that in SDP, it was implementation leaking ?

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt
Comment (KelvinKappeler) [inline app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt:58]: Why using a fake _FBCardPossession.NONE_ if you won't use it in _removeCardFromCollection_ ?

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt
Comment (KelvinKappeler) [inline app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt:40]: Why using FBMagicCard here? We don't care about the possession of a card.

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt
Comment (KelvinKappeler) [inline app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt:49]: Here too.

**Review by Cocograal — DISMISSED** [review]

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt
Comment (KelvinKappeler) [inline app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt:66]: Again same comment.

File: app/src/main/res/layout/fragment_single_card.xml
Comment (KelvinKappeler) [inline app/src/main/res/layout/fragment_single_card.xml:141]: You should use android:layout_marginEnd="10dp", not android:layout_marginRight

**Review by KelvinKappeler — CHANGES_REQUESTED** [review]
Good work in general, it should be discussed with the team, but I find that there are a lot of unnecessary parameters (having to add a possession by hand when it is useless for example).

In addition, the tests are sorely lacking (41% is not enough).

Il will wait for another review from our team :)

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardFragment.kt
Comment (annierusu) [reply app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardFragment.kt:132]: The argument in this function is an observer (line 40) of isConnected inside the viewmodel, which does have access to the connectivity of the user. Can you elaborate what you mean?

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt
Comment (annierusu) [reply app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsViewModel.kt:58]: yes, I just realised kotlin can have default arguments, which we should use, we keep every argument type as an FB card to keep the database code consistent.

**Review by jeamigue — DISMISSED** [review]
Some improvements could be made as pointed out by the other reviewer. I suggest we merge and create a new branch to improve the adressed issues, so we don't have big changes pending for too long, since nothing really blocking. Perfect should not always be the target.

File: app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt
Comment (annierusu) [reply app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt:40]: We use FBMagicCard in all the methods of the database, this is to keep the code consistent.

**Review by KelvinKappeler — APPROVED** [review]
Thanks for your changes.

**Pull request merged**
